### DISCUSSION TITLE
Can Livewire dispatch an event if validation fails?

### DISCUSSION THREAD

**Discussion by adriansio** [livewire/livewire#7460] [1. Help]
I don't want to use custom validators, I'm totally fine by using Livewire native validation with `$this->validate()`.

The issue is that if validation fails, I have no idea how I can run a little Javascript code (to show a global toaster saying "Hey there are some errors, please fix them").

This would be VERY easy if Livewire could dispatch an event if validation fails. In this way, I could simply listen for the event with Alpine and execute the Javascript code.

I've tried adding this to the component blade but it doesn't work at all, for some reasons it's executed only once:

```
@if ($errors->any())
    console.log('Yes, there are errors')
@else
    console.log('No errors, all good')
@endif
```

Is there really no way to execute JavaScript code when Livewire's own validation fails?

**Comment by sagor110090** [comment]
you should use laravel

$validator = Validator::make($request->all(), [
'title' => 'required|unique:posts|max:255',
'body' => 'required',
]);

if ($validator->fails()) {
return redirect('post/create')
->withErrors($validator)
->withInput();
}

**Comment by ghanu119** [comment]
@adriansio 
Hi,

You can do it by using the `boot` method of the component where you have to check the is validator fails or not and you can set up by using [link](https://livewire.laravel.com/docs/validation#accessing-the-validator-instance)

Then inside it you can write the your toast code. That explain in this Question: Check [`$this->js`](https://github.com/livewire/livewire/discussions/6295#discussion-5505438)

Or You can dispatch the direct event of Alpine from Livewire. Whatever you can use.

**Comment by chrisrhymes** [comment]
I have used a try/catch in the past to dispatch a custom event when validation fails. 

```php
try {
    $this->validate();
} catch (Throwable $e) {
    $this->dispatch('validationError');
    throw $e;
}
```

Then listen for the event in your JavaScript with `Livewire.on()` or maybe `$wire.on()`.

**Reply by madurapa** [reply]
`\Illuminate\Validation\ValidationException $e` more convenient than `Throwable $e` I think

**Comment by Fluotonic** [comment]
You can also do it this way if you want to stay on the same page :

```php
$this->withValidator(function (Validator $validator) {
      if($validator->fails()) {
           session()->flash('error', $validator->errors());
      }
})->validate();

```
Hope this helps!

**Comment by ghost** [comment]
This worked for me:


 ```
   public function boot()
    {
        $this->withValidator(function ($validator) {
            $validator->after(function ($validator) {
                if ($validator->errors()->count() > 0) {
                    $this->dispatch('flasher.error', message: 'Not saved. Please check for errors.');
                }
            });
        });
    }
```

https://livewire.laravel.com/docs/validation#accessing-the-validator-instance

**Reply by ghabriel25** [reply]
@jeremy-at-evosus Don't complicate things, you can dispatch it from `exception()` hook
```php
public function exception($e, $stopPropagation): void
{
    if ($e instanceof ValidationException) {
        $this->dispatch(...);
        
        $stopPropagation();
    }
}

**Reply by jeremykenedy** [reply]
I mean my solution is pulled directly from the live wire documentation 🤷‍♂️
